### DISCUSSION TITLE
Send slash commands (/something)

### DISCUSSION THREAD

**Discussion by minhla** [rigwild/discord-self-bot-console#31] [Q&A] [answered]
Hi, Discord will force all bots to use slash commands starting this 31st August, therefore I want to ask is it possible to add the ability to trigger a slash command of a bot in a channel of a server?

**Comment by rigwild** [comment, marked as answer]
Updated the script with `searchSlashCommand` and `sendSlashCommand`! ☺️

Example to search for a command named `/rank` and run it (with [the MEE6 bot](https://mee6.xyz/)):

```js
{
  id()

  const guildId = gid
  const channelId = cid

  // Search a command named `rank`, pick the first result
  const searchRes = await api.searchSlashCommand(channelId, 'rank')
  const command = searchRes.application_commands[0]

  // Get my rank, no command options
  await api.sendSlashCommand(guildId, channelId, command)

  // Get the rank of someone, pass the user in command options
  await api.sendSlashCommand(guildId, channelId, command, [{ type: 6, name: 'member', value: '1234535821234567890' }])
}
```

**Note:** The command search may return multiple `rank` commands, select the good one

If you are not sure what options you need to pass, simply open the Network tab of the devtools, manually send the command, then inspect the request Form Data (command options are in `payload_json.data.options`)


*Please mark as answer*

**Reply by minhla** [reply]
This works great! Thank you for the great work 😄

**Reply by minhla** [reply]
@rigwild Hello, would it be possible to modify the bot's function to be able to use slash command directly in the DM with the bot itself? Right now it returns this error of undefined:
```
  const searchRes = await api.searchSlashCommand(channelId, 'daily')
  const command = searchRes.application_commands[0]

----------------------------------------------

Uncaught TypeError: Cannot read property 'application_id' of undefined
    at Object.sendSlashCommand 
````
